openapi3/loader_recursive_ref_test.go

```diff
@@ -14,6 +15,44 @@ func TestLoaderSupportsRecursiveReference(t *testing.T) {
 	err = doc.Validate(loader.Context)
 	require.NoError(t, err)
 	require.Equal(t, "bar", doc.Paths["/foo"].Get.Responses.Get(200).Value.Content.Get("application/json").Schema.Value.Properties["foo2"].Value.Properties["foo"].Value.Properties["bar"].Value.Example)
+	require.Equal(t, "ErrorDetails", doc.Paths["/foo"].Get.Responses.Get(400).Value.Content.Get("application/json").Schema.Value.Title)
+	require.Equal(t, "ErrorDetails", doc.Paths["/double-ref-foo"].Get.Responses.Get(400).Value.Content.Get("application/json").Schema.Value.Title)
+}
+
+func TestResolvePath(t *testing.T) {
+	// relative path of folder
+	var b = &url.URL{Path: "testdata/recursiveRef"}
+	var c = &url.URL{Path: "./components/models/error.yaml"}
+	u, err := resolvePath(b, c)
+
+	require.NoError(t, err)
+	require.Equal(t, &url.URL{Path: "testdata/recursiveRef/components/models/error.yaml"}, u)
+
+	// relative path of specification file
+	b = &url.URL{Path: "testdata/recursiveRef/openapi.yaml"}
+	c = &url.URL{Path: "./components/models/error.yaml"}
+	u, err = resolvePath(b, c)
+
+	require.NoError(t, err)
+	require.Equal(t, &url.URL{Path: "testdata/recursiveRef/components/models/error.yaml"}, u)
+
+	// URL point to root of the resource
+	b, err = url.Parse("https://testdata/recursiveRef")
+	c = &url.URL{Path: "./components/models/error.yaml"}
+	u, err = resolvePath(b, c)
+
+	require.NoError(t, err)
+	var expected, _ = url.Parse("https://testdata/recursiveRef/components/models/error.yaml")
+	require.Equal(t, expected, u)
+
+	// URL point to the resource
+	b, err = url.Parse("https://testdata/recursiveRef/openapi.yaml")
+	c = &url.URL{Path: "./components/models/error.yaml"}
+	u, err = resolvePath(b, c)
+
+	require.NoError(t, err)
+	expected, _ = url.Parse("https://testdata/recursiveRef/components/models/error.yaml")
```

Comment:
Please always check for errors and keep the `NoError` call closer to the error creation.